contracts/Front/DmpBasedDesignator.sol

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright: Ambrosus Inc.
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+pragma solidity ^0.4.23;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "../Configuration/Config.sol";
+import "../Configuration/Time.sol";
+import "../Lib/SafeMathExtensions.sol";
+import "../Lib/DmpAlgorithm.sol";
+import "../Boilerplate/Head.sol";
+import "../Middleware/Sheltering.sol";
+import "../Storage/AtlasStakeStore.sol";
+
+
+contract DmpBasedDesignator is Base {
+
+    using SafeMath for uint;
+    using SafeMath for uint32;
+    using SafeMath for uint64;
+    using SafeMathExtensions for uint;
+
+    Time internal time;
+    Sheltering internal sheltering;
+    AtlasStakeStore internal atlasStakeStore;
+    Config internal config;
+
+    constructor(Head _head, Time _time, Sheltering _sheltering, AtlasStakeStore _atlasStakeStore, Config _config)
+    internal Base(_head)
+    {
+        time = _time;
+        sheltering = _sheltering;
+        atlasStakeStore = _atlasStakeStore;
+        config = _config;
+    }
+
+    function() public payable {}
+
+    function getRequestedBundle(bytes32 requestId) public view returns (bytes32);
+    function computeRequestDmpBaseHash(bytes32 requestId) public view returns (bytes32);
+    function getRequestCreationTime(bytes32 requestId) public view returns (uint64);
+    function requestIsInProgress(bytes32 requestId) public view returns (bool);
+
+    function canResolve(address resolverId, bytes32 requestId) public view returns (bool) {
+        bytes32 bundleId = getRequestedBundle(requestId);
+
+        // solium-disable-next-line operator-whitespace
+        return requestIsInProgress(requestId) &&
+        !sheltering.isSheltering(bundleId, resolverId) &&
+        resolverId == getRequestDesignatedShelterer(requestId) &&
+        atlasStakeStore.getStake(resolverId) > 0;
```

Comment:
Putting `canResolve` changes functionality for `ShelteringTransfers` contract. Specifically, it didn't check `atlasStakeStore.getStake(resolverId) > 0` condition before. I would say it makes sense to make sure the resolver still has stake when he wants to resolve a transfer. So the change itself is ok. But please write an additional test that covers this case for `ShelteringTransfers` contract.

Reply:
Or maybe we can remove `canResolve` from the superclass, as it's too loosely connected to DMP.

Reply:
Remove canResolve from the superclass?